shim/linux/src/entry.c

```diff
@@ -826,10 +827,24 @@ dispatch_vcpu_kvm_get_one_reg(struct kvm_one_reg *const ioctl_args)
 }
 
 static long
-dispatch_vcpu_kvm_get_regs(struct kvm_regs *const ioctl_args)
+dispatch_vcpu_kvm_get_regs(
+    struct shim_vcpu_t const *const pmut_vcpu,
+    struct kvm_regs *const pmut_ioctl_args)
 {
-    (void)ioctl_args;
-    return -EINVAL;
+    struct kvm_regs mut_args;
+
+    if (handle_vcpu_kvm_get_regs(pmut_vcpu, &mut_args)) {
+        bferror("handle_vcpu_kvm_get_regs failed");
+        return -EINVAL;
+    }
+
+    platform_copy_to_user(pmut_ioctl_args, &mut_args, sizeof(struct kvm_regs));
+    if (NULL == pmut_ioctl_args) {
```

Comment:
Remove this if statement. If this is true, the kernel would segfault from a NULL dereference when platform_copy_to_user is executed. platform_copy_to_user has a void return, so there is nothing to validate here.